DifferentiationInterface/ext/DifferentiationInterfaceGTPSAExt/onearg.jl

```diff
@@ -0,0 +1,634 @@
+    if D != Nothing
+        d = backend.descriptor
+    else
+        d = Descriptor(1, 1) # 1 variable to first order
+    end
+    if x isa Number
+        xt = TPS{promote_type(typeof(first(tx)), typeof(x), Float64)}(; use=d)
+        return GTPSAOneArgPushforwardPrep(xt)
+    else
+        xt = similar(x, TPS{promote_type(eltype(first(tx)), eltype(x), Float64)})
+        for i in eachindex(xt)
+            xt[i] = TPS{promote_type(eltype(first(tx)), eltype(x), Float64)}(; use=d)
+        end
+        return GTPSAOneArgPushforwardPrep(xt)
+    end
+end
+
+function DI.pushforward(
+    f,
+    prep::GTPSAOneArgPushforwardPrep,
+    ::AutoGTPSA,
+    x,
+    tx::NTuple,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    fc = DI.with_contexts(f, contexts...)
+    ty = map(tx) do dx
+        foreach((t, xi, dxi) -> (t[0] = xi; t[1] = dxi), prep.xt, x, dx)
+        yt = fc(prep.xt)
+        if yt isa Number
+            return yt[1]
+        else
+            dy = map(t -> t[1], yt)
+            return dy
+        end
+    end
+    return ty
+end
+
+function DI.pushforward!(
+    f,
+    ty::NTuple,
+    prep::GTPSAOneArgPushforwardPrep,
+    ::AutoGTPSA,
+    x,
+    tx::NTuple,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    fc = DI.with_contexts(f, contexts...)
+    for b in eachindex(tx, ty)
+        dx, dy = tx[b], ty[b]
+        foreach((t, xi, dxi) -> (t[0] = xi; t[1] = dxi), prep.xt, x, dx)
+        yt = fc(prep.xt)
+        map!(t -> t[1], dy, yt)
+    end
+    return ty
+end
+
+function DI.value_and_pushforward(
+    f,
+    prep::GTPSAOneArgPushforwardPrep,
+    backend::AutoGTPSA,
+    x,
+    tx::NTuple,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    fc = DI.with_contexts(f, contexts...)
+    ty = DI.pushforward(fc, prep, backend, x, tx)
+    y = fc(x) # TO-DO: optimize
+    return y, ty
+end
+
+function DI.value_and_pushforward!(
+    f,
+    ty::NTuple,
+    prep::GTPSAOneArgPushforwardPrep,
+    backend::AutoGTPSA,
+    x,
+    tx::NTuple,
+    contexts::Vararg{DI.Context,C},
+) where {C}
+    fc = DI.with_contexts(f, contexts...)
+    DI.pushforward!(fc, ty, prep, backend, x, tx)
+    y = fc(x)  # TO-DO: optimize
+    return y, ty
+end
+
+## Gradient
+# Contains either a single pre-allocated initial TPS
+# or a vector of pre-allocated TPSs.
+struct GTPSAOneArgGradientPrep{X} <: DI.GradientPrep
+    xt::X
+end
+
+# Unlike JVP, this requires us to use all variables 
+function DI.prepare_gradient(
+    f, backend::AutoGTPSA{D}, x, contexts::Vararg{DI.Context,C}
+) where {D,C}
+    if D != Nothing
+        d = backend.descriptor
+    else
+        d = Descriptor(length(x), 1) # n variables to first order
+    end
+
+    # We set the slopes of each variable to 1 here, this will always be the case for gradient
+    if x isa Number
+        xt = TPS{promote_type(typeof(x), Float64)}(; use=d)
+        xt[1] = 1
+        return GTPSAOneArgGradientPrep(xt)
```

Comment:
`gradient` cannot be called on `x::Number`, hence the lack of coverage for these lines (which can be deleted)